src/test/lib/EventTest.cpp

```diff
@@ -557,6 +584,205 @@ void QuicTestValidateConnectionEvents(uint32_t Test)
     } // Listener Scope
 }
 
+#if defined(QUIC_API_ENABLE_PREVIEW_FEATURES)
+void
+QuicTestValidateNetStatsConnEvent1(
+    _In_ MsQuicRegistration& Registration,
+    _In_ MsQuicListener& Listener,
+    _In_ QuicAddr& ServerLocalAddr
+    )
+{
+    TestScopeLogger ScopeLogger(__FUNCTION__);
+
+    MsQuicSettings Settings;
+    Settings.SetMinimumMtu(1280).SetMaximumMtu(1280);
+    Settings.SetCongestionControlAlgorithm(QUIC_CONGESTION_CONTROL_ALGORITHM_BBR);
+    Settings.SetNetStatsEventEnabled(true);
+    MsQuicConfiguration ServerConfiguration(Registration, "MsQuicTest", Settings, ServerSelfSignedCredConfig);
+    TEST_TRUE(ServerConfiguration.IsValid());
+
+    MsQuicConfiguration ClientConfiguration(Registration, "MsQuicTest", Settings, MsQuicCredentialConfig());
+    TEST_TRUE(ClientConfiguration.IsValid());
+
+    ConnValidator Client(
+        new(std::nothrow) ConnEventValidator* [4] {
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_DATAGRAM_STATE_CHANGED),
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_CONNECTED, QUIC_EVENT_ACTION_SHUTDOWN_CONNECTION),
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE),
+            nullptr
+        }
+    );
+    ConnValidator Server(
+        new(std::nothrow) ConnEventValidator* [4] {
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_CONNECTED, 0, true),
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_SHUTDOWN_INITIATED_BY_PEER),
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_SHUTDOWN_COMPLETE),
+            nullptr
+        },
+        ServerConfiguration
+    );
+
+    Client.SetEventsExpectedAtleastOnce(
+        new(std::nothrow) ConnEventValidator* [2] {
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_NETWORK_STATISTICS),
+	    nullptr
+        }
+    );
+
+    Server.SetEventsExpectedAtleastOnce(
+        new(std::nothrow) ConnEventValidator* [2] {
+            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_NETWORK_STATISTICS),
+	    nullptr
```

Comment:
```suggestion
            new(std::nothrow) ConnEventValidator(QUIC_CONNECTION_EVENT_NETWORK_STATISTICS),
	        nullptr
```
Please fix others too